packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -0,0 +1,156 @@
+        if (Array.isArray(currentBrowserFeatureSupported) && currentBrowserFeatureSupported.length > 0) {
+            if (prefix) {
+                currentBrowserFeatureSupported = currentBrowserFeatureSupported.find((info) => {
+                    return info.prefix === prefix;
+                });
+            } else {
+                currentBrowserFeatureSupported = currentBrowserFeatureSupported.find((info) => {
+                    return !info.prefix;
+                });
+            }
+        }
+
+        return currentBrowserFeatureSupported as SimpleSupportStatement;
+    }
+
+    /* eslint-disable camelcase */
+    public getWorstCaseSupportStatementFromInfo(browserFeatureSupported: SupportStatement | undefined): SimpleSupportStatement | undefined {
+        // If we dont have information about the compatibility, ignore.
+        if (!browserFeatureSupported) {
+            return browserFeatureSupported;
+        }
+
+        // Take the smaller version_removed and bigger version_added
+        const worstBrowserFeatureSupported: SimpleSupportStatement = {
+            version_added: null,
+            version_removed: null
+        };
+
+        if (Array.isArray(browserFeatureSupported) && browserFeatureSupported.length > 0) {
+            browserFeatureSupported.forEach((info) => {
+                if (!worstBrowserFeatureSupported.version_added && info.version_added === true) {
+                    worstBrowserFeatureSupported.version_added = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_added || worstBrowserFeatureSupported.version_added && info.version_added && info.version_added > worstBrowserFeatureSupported.version_added) {
+                    worstBrowserFeatureSupported.version_added = info.version_added;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed && info.version_removed === true) {
+                    worstBrowserFeatureSupported.version_removed = true;
+                }
+
+                if (!worstBrowserFeatureSupported.version_removed || worstBrowserFeatureSupported.version_removed && info.version_removed && info.version_removed < worstBrowserFeatureSupported.version_removed) {
+                    worstBrowserFeatureSupported.version_removed = info.version_removed;
+                }
+            });
+
+            return worstBrowserFeatureSupported;
+        }
+
+        return browserFeatureSupported as SimpleSupportStatement;
+    }
+    /* eslint-enable camelcase */
+
+    private isFeatureRequiredToTest(typedFeatureValue: CompatStatement & MDNTreeFilteredByBrowsers, isCheckingNotBroadlySupported = false): boolean {
+        // TODO: Here we are checking only parent but this object has children
+        let isRequiredToTest = false;
+
+        forEach(this.browsers, (browserVersionsList, browser) => {
+            if (isRequiredToTest || !typedFeatureValue.__compat || !typedFeatureValue.__compat.support) {
```

Comment:
I'd do something similar to what I proposed in my previous comment about using `every` to validate everything so it's easier to know what we are validating.